### ISSUE TITLE
Create function to parse a hex string into a byte array (`TCMD_extract_hex_array_arg`)

### ISSUE THREAD

**Issue by DeflateAwning** [CalgaryToSpace/CTS-SAT-1-OBC-Firmware#89]
* Needed for Issue #32.
* Needed to finish implementing the `TCMDEXEC_flash_write_hex` function, which currently throws an annoying build warning

```c
uint8_t TCMD_extract_hex_array_arg(const char *args_str, uint8_t arg_index, uint8_t result_array[], uint16_t result_array_size);
```

Function should support ignoring spaces and underscores within the string, which are used as delimeters. 

You may wish to factor out the hex parsing logic to one or more other functions, to separate it from the comma-searching nature of the `TCMD_extract_` functions.

Requires a unit test.

### Examples

* Invalid: `0x0000` -> do not allow "0x" prefix
* Valid: `0123FF` -> simple example case, goes into length-3 byte array `{0x01, 0x23, 0xFF}` (which is the same as `{1, 35, 255}`)
* Valid: `01` -> `{0x01}`
* Valid: `01_02_03` -> `{1, 2, 3}` (underscore can be a delimeter between bytes)
* Invalid: `010_203` -> underscore/space cannot be a delimeter in the middle of bytes (must be between even-numbered hex chars)
* Invalid: `1`, `12345` -> must be an even length
* Invalid: `01_02_3` -> must be an even length

**Comment by robotoshi** [comment]
This is a duplicate of #77, but with more details

**Comment by parker-research** [comment]
Implemented in #111
